### ISSUE TITLE
Automatic sync for webdav

### ISSUE THREAD

**Issue by Nebucatnetzer** [orgzly/orgzly-android#625]
I love the integration of webdav.
However is there a way to automatically sync the notebooks?
Or is there an intent so that I can trigger it from Tasker?

```
am startservice --user 0 -a com.orgzly.intent.action.SYNC_START com.orgzly/com.orgzly.android.sync.SyncService
```
Doesn't work.

**Comment by NiceFeather** [comment]
Sure, that's a good idea, just like ownCloud desktop client.

**Comment by nevenz** [comment]
It's part of #8 as @NiceFeather said. Auto-sync is restricted to Directory repository type as there were issues with Dropbox syncing. It was too easy to hit their API way too much.

There is a plan to queue syncing requests (I plan to start playing with WorkManager for that soon), but perhaps this should be expedited as I assume a lot of WebDAV users will be using a local storage anyway (like Nextcloud), myself included.

Perhaps a per-repository option could be added to control auto-syncing.

> am startservice --user 0 -a com.orgzly.intent.action.SYNC_START com.orgzly/com.orgzly.android.sync.SyncService

You might need to use `start-foreground-service`.

**Comment by Nebucatnetzer** [comment]
I wasn't able to get it working with the shell command in Tasker. It always showed an error even though the command worked with adb shell.
I implemented now this: https://github.com/orgzly/orgzly-android/issues/8#issuecomment-466646362 and it's working.
I don't know if we want to close this issue in favour of #8 since my problem is now solved and #8 basically covers everything already?

**Comment by nevenz** [comment]
> I implemented now this: #8 (comment) and it's working.

Ah yes, `com.orgzly.android.ActionReceiver` was actually created for this and is preferable instead of using `SyncService` directly.

> I don't know if we want to close this issue in favour of #8 since my problem is now solved and #8 basically covers everything already?

Sounds good, I'll tag it as a duplicate.
